models/issues/issue.go

```diff
@@ -1077,19 +1077,38 @@ func NewIssue(repo *repo_model.Repository, issue *Issue, labelIDs []int64, uuids
 	}
 	defer committer.Close()
 
+	// onFail will try to gracefully revert the update in the resource index.
+	onFail := func() error {
+		// Close commiter.
+		committer.Close()
+		// Try to revert the increase in resource index.
+		if err := db.UpsertDecrResourceIndex(db.DefaultContext, "issue_index", repo.ID, idx); err != nil {
+			return fmt.Errorf("UpsertDecrResourceIndex: %v", err)
+		}
+		return nil
+	}
+
 	if err = NewIssueWithIndex(ctx, issue.Poster, NewIssueOptions{
 		Repo:        repo,
 		Issue:       issue,
 		LabelIDs:    labelIDs,
 		Attachments: uuids,
 	}); err != nil {
+		if err := onFail(); err != nil {
+			return fmt.Errorf("couldn't gracefully handle error: %v", err)
```

Comment:
```suggestion
			return fmt.Errorf("couldn't gracefully handle error: %w", err)
```

Reply:
Is this a new code-style I haven't been aware of? Since when are we going to use the wrapping errors verb?

Reply:
Well, as I've learned: go seems to do some magic if you use `%w` instead of `%v`:
https://stackoverflow.com/a/61287626

Reply:
if we can wrap stuff my pr which tries to create a common hierarchy of errors will be a lot easier